server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2314,7 +2317,7 @@ public void openEngineAndRecoverFromTranslog() throws IOException {
         };
 
         // Do not load the global checkpoint if this is a remote snapshot index
-        if (indexSettings.isRemoteSnapshot() == false) {
+        if (indexSettings.isRemoteSnapshot() == false && indexSettings.isRemoteTranslogStoreEnabled() == false) {
```

Comment:
nit: We can use the same check that we used in `trimTranslog` - `isRemoteTranslogEnabled()`